module/os/linux/zfs/vdev_file.c

```diff
@@ -139,15 +153,13 @@ vdev_file_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
 	 */
 	ASSERT(vd->vdev_path != NULL && vd->vdev_path[0] == '/');
 
-	error = zfs_file_open(vd->vdev_path,
```

Comment:
I don't think it's a good idea to change the original sync implementation to async one. Instead, we'd better use an increamental or configurable way, say, add an async implementation for vdev file.